znframe/frame.py

```diff
@@ -40,12 +43,39 @@ class Frame:
     positions: np.ndarray = field(
         converter=_list_to_array, eq=cmp_using(np.array_equal)
     )
-    arrays: dict[str, np.ndarray] = field(converter=_list_to_array, eq=False)
+
+    connectivity: np.ndarray = field(
+        converter=_list_to_array, eq=cmp_using(np.array_equal), default=None
+    )
+
+    arrays: dict[str, np.ndarray] = field(
+        converter=_list_to_array, eq=False, factory=dict
+    )
     info: dict[str, float | int | np.ndarray] = field(
-        converter=_list_to_array, eq=False
+        converter=_list_to_array, eq=False, factory=dict
+    )
+
+    pbc: np.ndarray = field(
+        converter=_list_to_array,
+        eq=cmp_using(np.array_equal),
+        default=np.array([True, True, True]),
     )
-    pbc: np.ndarray = field(converter=_list_to_array, eq=cmp_using(np.array_equal))
-    cell: np.ndarray = field(converter=_cell_to_array, eq=cmp_using(np.array_equal))
+    cell: np.ndarray = field(
+        converter=_cell_to_array, eq=cmp_using(np.array_equal), default=np.zeros(3)
+    )
+
+    def __attrs_post_init__(self):
+        if self.connectivity is None:
+            ase_bond_calculator = ASEComputeBonds()
+            self.connectivity = ase_bond_calculator.build_graph(self.to_atoms())
+            self.connectivity = ase_bond_calculator.get_bonds(self.connectivity)
```

Comment:
Could this be a converter instead of being part of the `__post_init__`

Reply:
This is not simply done as you need an `ase.Atoms`-object to calculate the bonds. And it is not possible to convert the `frame` to an `ase.Atoms` while the initialization is still in progess.